web/approval.go

```diff
@@ -297,15 +310,40 @@ func calculateApprovalInfo(request *model.ApprovalRequest, policy *model.Approva
 	if err != nil {
 		return nil, err
 	}
-	approved = approved && audit && authorAffirm
+	if !approved {
+		// only after sufficient approvals are received
+		// test the author affirm feature
+		authAffirm = true
+	} else {
+		authAffirm = authorAffirm(request, policy)
+		if !authAffirm {
+			clone := &(*request)
+			for _, c := range request.Commits {
+				removeAuthor(c.Author, clone)
+			}
+			approvers = set.Empty()
+			authAffirm, err = model.Approve(clone, policy,
```

Comment:
I think what's confusing is that line 325 is assigning to `authAffirm` instead of `approved`. It's on purpose but possibly the code could be refactored to make it read better.